contracts/src/messaging/messenger/CrossChainEnabledObscuro.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./ICrossChainMessenger.sol";
+
+// TODO: We need to upgrade the open zeppelin version to the 4.x that adds cross chain enabled
+abstract contract CrossChainEnabledObscuro {
+    
+    ICrossChainMessenger messenger;
+    IMessageBus messageBus;
+    uint32 nonce = 0;
+
+    // The messenger contract passed will be the authority that we trust to tell us
+    // who has sent the cross chain message and that the message is indeed cross chain.
```

Comment:
qq. this `messengerAddress` corresponds to the key held by the enclave alone?

Reply:
No, this is the smart contract that we trust to relay the messages. CrossChainMessenger.sol is said messenger.

Reply:
right, the message bus, who in turn is driven by the platform enclave